USFMToolsSharp/Models/Markers/FMarker.cs

```diff
@@ -36,6 +36,14 @@ public override string PreProcess(string input)
             typeof(WMarker),
             typeof(WEndMarker),
             typeof(TextBlock),
+            typeof(UnknownMarker),

```

Comment:
@rbnswartz 
The problem that we are running across is that we are finding LOTS of USFM Bibles that have tags that are outside of the standard 3.0 set.  Thus far, they have been located within the footnotes.  I believe, but could be wrong, that these are related to unofficial tags that Paratext uses for their enhanced resources.

Example:

```
\v 7 Solomon the father of Rehoboam, Rehoboam the father of Abijah, Abijah the father of Asa, \f + \fr 1:7 \fn 4\fn* \fl Textual Criticism\fl* \ft The reading \grk ᾿Ασάφ\grk* (\tgrk Asaph\tgrk*), a variant spelling on \grk ᾿Ασά\grk* (\tgrk Asa\tgrk*), is found in the earliest and most widespread witnesses (𝔓\sup 1vid\sup* \heb א\heb* B C [D\sup luc\sup*] \it f\it* \sup 1, 13\sup* 700 \it pc\it* it co). Although Asaph was a psalmist and Asa was a king, it is doubtful that the author mistook one for the other since other ancient documents have variant spellings on the king's name (such as “Asab,” “Asanos,” and “Asaph”). Thus the spelling \grk ᾿Ασάφ\grk* that is almost surely found in the original of \fx Mt 1:7-8\fx* has been translated as “Asa” in keeping with the more common spelling of the king's name.\f*
```
```
v 16 and Jacob the father of Joseph, the husband of Mary, by whom \f + \fr 1:16 \fn 9\fn* \fl Textual Criticism\fl* \ft There are three significant variant readings at this point in the text. Some \sc mss\sc* and versional witnesses (\grk Θ\grk* \it f\it* \sup 13\sup* it) read, “Joseph, to whom the virgin Mary, being betrothed, bore Jesus, who is called Christ.” This reading makes even more explicit than the feminine pronoun (see \fl Study Note \fl* below) the virginal conception of Jesus and as such seems to be a motivated reading. The Sinaitic \ver Syriac\ver* ms alone indicates that Joseph was the father of Jesus (“Joseph, to whom was betrothed Mary the virgin, fathered Jesus who is called the Christ”). Although much discussed, this reading has not been found in any Greek witnesses. B. M. Metzger suggests that it was produced by a careless scribe who simply reproduced the set formula of the preceding lines in the genealogy (\it TCGNT\it* 6). In all likelihood, the two competing variants were thus produced by intentional and unintentional scribal alterations respectively. The reading adopted in the translation has overwhelming support from a variety of witnesses (𝔓\sup 1\sup* \heb א\heb* B C L W [\it f\it* \sup 1\sup*] 33 𝔐 co), and therefore should be regarded as authentic. For a detailed discussion of this textual problem, see \it TCGNT\it* 2-6. \fp \fl Study Note \fl* The pronoun \it whom\it* is feminine gender in the Greek text, referring to Mary.\f* Jesus was born, who is called Christ. \f + \fr 1:16 \fn 10\fn* \fl Translator's Note\fl* \ft Or “Messiah”; both “Christ” (Greek) and “Messiah” (Hebrew and Aramaic) mean “one who has been anointed.” \fp \fl Study Note \fl* The term \grk χριστός\grk* (\tgrk christos\tgrk*) was originally an adjective (“anointed”), developing in \ver LXX\ver* into a substantive (“an anointed one”), then developing still further into a technical generic term (“the anointed one”). In the intertestamental period it developed further into a technical term referring to the hoped-for anointed one, that is, a specific individual. In the NT the development starts there (technical-specific), is so used in the gospels, and then develops in Paul to mean virtually Jesus' last name.\f*
```


As the library currently stands, when it hits these types of tags, it mucks up the completion of creating the verse.  While yes, it does indicate an error, it isn't helpful.  I don't believe we want to get in the habit of adding in all these non-standard tags to the library.  By adding in the Unknown marker as acceptable to the Verse tag and the combination of the UnknownEnd tag, we can then wrap up these non-standard tags and, more importantly, allow the verse to encapsulate the tag fully and not blow up the verse parsing.

For our purposes, we are only trying to pull out the verse text while dropping all the footnote, image, etc tags.  This method accomplishes that for us.   If you have any other ideas of how to "ignore" these non-standard tags so they don't cause the verse parsing to fail, I'm all ears.
  If we parse these

Reply:
Yeah I think lets just add a "Ignore unknown markers" option for the parser. Unknown Marker was meant to signify an issue in the USFM but the more I think about it, in a production scenario ignoring unknowns is much better.